CRM/Civicase/Service/CaseRoleCreationPostProcess.php

```diff
@@ -160,21 +160,13 @@ private function getOtherActiveExistingRelationships($relTypeId, $caseId, $curre
    *   API request parameters.
    */
   private function setRelationshipsInactive(array $relIds, array $params) {
-    $endDate = empty($params['end_date']) ? new DateTime() : new DateTime($params['end_date']);
-    $today = new DateTime();
-    $isEndDateSameAsToday = $endDate->format('Y-m-d') === $today->format('Y-m-d');
-    $relationshipActiveFields = $isEndDateSameAsToday
-      ? ['is_active' => 0]
-      : ['end_date' => $params['end_date']];
-
     foreach ($relIds as $relId) {
-      civicrm_api3('Relationship', 'create', array_merge(
-        $relationshipActiveFields,
-        [
-          'id' => $relId,
-          'skip_post_processing' => 1,
-        ]
-      ));
+      civicrm_api3('Relationship', 'create', [
+        'id' => $relId,
+        'is_active' => 0,
+        'end_date' => !empty($params['start_date']) ? $params['start_date'] : date('Y-m-d'),
```

Comment:
why are we using start date param for end date here also previously this method was being used for setting is_active to 0 or end date only so can we make sure that it is working as expected for other areas that refers to this method

Reply:
Thanks @shahrukh-compucorp, this functionality was introduced in this PR:
https://github.com/compucorp/uk.co.compucorp.civicase/commit/e8377a4a45d6ac5fdd30f8e84615c2e9bf927a3a#diff-47d043a7132a7d4bf0d2bc9d374de22b30769646ee53efa0d155d585875f3a8bR167

We are just reverting the function how it previously was before the https://github.com/compucorp/uk.co.compucorp.civicase/pull/626 change.

I have refactored the function a bit so we pass the end date directly and not all the request parameter.

To explain what this is doing a bit better: this function sets all previous case relationship as inactive when assigning a new one. Using the "start date" as the "end date" means that we are using the "start date" of the new relationship as the "end date" of the previous relationships.